src/init.cpp

```diff
@@ -1952,6 +1984,31 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
 
     connOptions.m_i2p_accept_incoming = args.GetBoolArg("-i2pacceptincoming", DEFAULT_I2P_ACCEPT_INCOMING);
 
+    if (args.GetBoolArg("-privatebroadcast", DEFAULT_PRIVATE_BROADCAST)) {
+        // If -listenonion is set, then NET_ONION may not be reachable now
+        // but may become reachable later, thus only error here if it is not
+        // reachable and will not become reachable for sure.
+        const bool onion_may_become_reachable{listenonion && (!args.IsArgSet("-onlynet") || onlynet_used_with_onion)};
+        if (!g_reachable_nets.Contains(NET_I2P) &&
+            !g_reachable_nets.Contains(NET_ONION) &&
+            !onion_may_become_reachable) {
+            return InitError(_("Private broadcast of own transactions requested (-privatebroadcast), "
```

Comment:
Emitting errors from the RPC is also an option, but I prefer to keep the current way because it clearly shows something is misconfigured and starting up knowing that the RPC will never work and always give errors seems like a bad user experience. I would prefer that the user stops and is forced to fix the misconfiguration of the node. `bitcoin-qt` also has a RPC console.

Reply:
ok thanks for the answer.
I also thought about that it might make sense to have an optional argument for _sendrawtransaction_ rpc, whether to use privatebroadcast or not (for example in the case when it cannot be used, but user still wants to broadcast without restarting bitcoind)
Same answer also applies to that I guess

Reply:
Yeah. Also, having an extra argument to the RPC is an alternative of the `-privatebroadcast` config option that would be maybe viable if the plan was to stop with the RPC, but having in mind that the wallet will eventually also use private broadcast, I would like to avoid having to configure each module separately for private broadcast, so a single global config option seems to fit best.

Reply:
makes sense, thanks

Reply:
This is unfortunate that `privatebroadcast` is incompatible with `connect`. The use case would be to connect to a single trusted peer for listening, but broadcast through ephemeral connections to not link the trusted peer to our wallet.

Reply:
By "for listening" you mean "for learning new transactions and blocks", right? So, in that setup the node is not listening (not accepting inbound connections), has only one connection to a known honest node and broadcasts its own transactions using private broadcast? That seems like an useful setup that would be nice to have support for. But the current semantic of `-connect` is "only connect to the specified node(s)" which is at odds with "connect to random nodes for transaction broadcast". I don't like changing the semantic of `-connect` like that.

I wonder if this can be supported and to keep the current meaning of `-connect`... don't use `-connect` but somehow restrict our node to be able to only connect to 1 address :thinking:

Reply:
But we are adding a new option here `privatebroadcast`. Could we not change the semantics of `connect` *if* the new option `privatebroadcast` is used?

Reply:
Yeah, you are right that `-connect=0 -privatebroadcast=1` did not exist before because `-privatebroadcast=` did not exist. Then the semantic of `-connect=0` would be "do this if privatebroadcast=0 and do something else if privatebroadcast=1". I do not like that, seems convoluted and a possible source of confusion to make the behavior of one option dependent on the value of another option. I would like to have less of that stuff, not more.

Reply:
I suppose I can `addnode` my trusted node for now, but I will be leaking other outbound connections.
I'm not sure how to handle my case unless we don't consider the ephemeral broadcast connections as true "connections" in the sense of `connect` option.

Reply:
I suppose I can `addnode` my trusted node for now, but I will be leaking other outbound connections.
I'm not sure how to handle my case unless we don't consider the ephemeral broadcast connections as true "connections" in the sense of `connect` option.

Reply:
Just an idea - if the trusted node is accepting inbound Tor connections, then run the private node with `-onlynet=onion -addnode=trusted.onion:8333 -privatebroadcast=1`. Benefit of connecting to the trusted node over Tor - you know that you actually connected to the trusted node (or somebody who owns the private key for the `trusted.onion` service). If you connect over clearnet to an IP address, then you have to use v2 (bip324) _and_ verify that the session ids match (`session_id` in `getpeerinfo` RPC output on both sides of the connection).

Reply:
This is a hack and I haven't tested it, but instead of using `-connect`,  `-maxconnections=0` `-privatebrodcast` and then addnode your trusted peer might work for your use case?

Reply:
> -maxconnections=0 -privatebrodcast and then addnode your trusted peer

Thanks for the idea, but sadly this does not work for this either. It will happily add only my trusted node, but it will not create the private broadcast connections when actually trying to send. It will just wait at 0 connections.

Since we break the maxconnections value already for the addnode exception, could we make an exception for private broadcast connections as well?

Reply:
Yes, that is the case:

> argsman.AddArg("-maxconnections=<n>", ... It does not apply to short-lived private broadcast connections either, which have a separate limit of MAX_PRIVATE_BROADCAST_CONNECTIONS

You can run with `-debug=privatebroadcast` and see what happens when you send a new transaction.